pde/storage/_ffmpeg.py

```diff
@@ -0,0 +1,79 @@
+"""
+Functions for interacting with FFmpeg
+
+.. codeauthor:: David Zwicker <david.zwicker@ds.mpg.de>
+"""
+
+# import subprocess as sp
+from dataclasses import dataclass
+
+import numpy as np
+from numpy.typing import DTypeLike
+
+# def _run_ffmpeg(args: list[str]):
+#     return sp.check_output(["ffmpeg"] + args)
+#
+#
+# def codecs() -> list[str]:
+#     """list: all supported ffmpeg codecs"""
+#     res = _run_ffmpeg(["-codecs"])
+#
+#
+# def get_pixel_formats(encoder=None):
+#     if encoder is None:
+#         res = _run_ffmpeg(["-pix_fmts"])
+#     else:
+#         res = _run_ffmpeg(["-h", f"encoder={encoder}"])
+
+
+@dataclass
+class FFmpegFormat:
+    """defines a FFmpeg format used for storing field data in a video"""
+
+    pix_fmt_file: str
+    codec: str
+    channels: int
+    value_max: int
+    dtype: DTypeLike
+
+    @property
+    def pix_fmt_data(self) -> str:
+        return {1: "gray", 3: "rgb24", 4: "rgba"}[self.channels]
```

Comment:
```suggestion
    def pix_fmt_data(self) -> str:
        if self.channels == 1:
            return "gray"
        elif self.channels == 3:
            return "rgb24"
        elif self.channels == 4:
            return "rgba"
        else:
            raise ValueError()
```

In this way it is much easier to understand what is going on.

Moreover, what happens if self.channels is not one of 1, 3, 4 ?

With your method: a `key-error` will be thrown but actually is a `value error` because the class work just if we have `self.channels` in [1, 3, 4]. 

Perhaps, an idea could be to add a check in the __post_init__ method to ensure that channels is one of the above one.

Reply:
Good point! I'll adjust it to throw a more useful error in the future.